### PR TITLE
extract hot system variables from map

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

### What problem does this PR solve? 

System variables "warning_count" and "error_count" are set for each statement here https://github.com/pingcap/tidb/blob/master/executor/executor.go#L1487
https://github.com/pingcap/tidb/blob/master/executor/executor.go#L1619
https://github.com/pingcap/tidb/blob/master/executor/executor.go#L1623

### What is changed and how it works?
Extract these two hot system variable from map to reduce the CPU cost.

### Check List 

Tests 

 - Unit test

Code changes

- No

Side effects
- No

Related changes

- No

Release note

 - No

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @coocood

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -765,8 +774,46 @@ func (s *SessionVars) WithdrawAllPreparedStmt() {
 	metrics.PreparedStmtGauge.Set(float64(afterMinus))
 }
 
+func (s *SessionVars) trySetWarningCountOrErrorCount(name string, val string) (bool, error) {
```
Comment (coocood) [inline sessionctx/variable/session.go]: We don't need to do this, `error_count` and `warning_count` is read-only

Comment (zhangjinpeng1987) [reply sessionctx/variable/session.go]: Thank you for letting me know that.

**Comment by coocood** [comment]
LGTM

**Pull request merged**